### ISSUE TITLE
Resolve errors and warnings from Eclipse oxygen 4.7.3

### ISSUE THREAD

**Issue by romani** [checkstyle/checkstyle#5916]
after import to Eclipse:

during import:
![image](https://user-images.githubusercontent.com/812984/41385526-0e8f992a-6f31-11e8-860f-71f8745d0a7e.png)

ERRORS and warnings are at 
https://gist.github.com/romani/2c5111fd8daf88d3524478c37278b196

1)
`Plugin execution not covered by lifecycle configuration: org.codehaus.mojo:exec-maven-plugin:1.6.0:java (execution: default, phase: process-classes)	pom.xml	/checkstyle	line 708	Maven Project Build Lifecycle Mapping Problem`

2) compilation problems in inputs

3) TODO in non related Inputs

4) warnings on deprecated method usage

5) `Access restriction: The type 'Constants' is not API (restriction on required library '/opt/jdk/jdk1.8.0_172/jre/lib/rt.jar')  XmlLoaderTest.java  /checkstyle/src/test/java/com/puppycrawl/tools/checkstyle line 41 Java Problem`

**Expected**: no problems and almost no warnings after following http://checkstyle.sourceforge.net/eclipse.html

**Comment by romani** [comment]
> Access restriction: The type 'Constants'

Looks like was added during pitest coverage:
https://github.com/checkstyle/checkstyle/blame/dac8dda2f923ad67a235dd58635a94f443141946/src/test/java/com/puppycrawl/tools/checkstyle/XmlLoaderTest.java , lets try to remove it, I do not see why it might required

**Comment by romani** [comment]
> warnings on deprecated method usage

this is unavoidable as we demand 100% coverage, need to be marked as "Ignore"  level in Eclipse.

**Comment by romani** [comment]
Looks like new eclipse do validate resources:
![image](https://user-images.githubusercontent.com/812984/41387244-baceb58c-6f3a-11e8-8291-9d59bc01d6c0.png)

we might need to completely remove resources from Eclipse build path . @rnveach , do you have any other suggestions.

**Comment by romani** [comment]
> The declared package "" does not match the expected package

just need to be fixed, where it is not part of test case. Test case probably should be moved to non-compilable

**Comment by rnveach** [comment]
> Looks like new eclipse do validate resources:

Eclipse has always validated resources, it is not new. The POM is set to compile resources and since Eclipse sees this, it thinks resources are part of the main code and should be compiled and displayed errors and such on them.

My eclipse will print problems on resources after it finishes compiling them, giving me a lot of useless errors. Just adding `test\resources` generates 6400 errors and 2700 warnings. I specifically go into Property > Java Build Path and remove `it\resources` and `test\resources` to prevent this. I have to do this every time Eclipse updates it settings based on the maven POM. Usually Eclipse wants me to do this update every time a new version of a dependency is added to the POM.

I have no other workaround besides changing the POM.

**Comment by romani** [comment]
> Eclipse has always validated resources, it is not new.

When we created such html page with Eclispe setup, there was no such problems. Smth is changed in few years. Warnings is not a big of a deal, but there should be no errors at all.

> I specifically go into Property > Java Build Path and remove it\resources and test\resources to prevent this

Can you review page http://checkstyle.sourceforge.net/eclipse.html and fix what else is missed to start work.

**Comment by romani** [comment]
One more strange problem, I had such problem ones on import of project, not it is reproduced again:
```
Description	Resource	Path	Location	Type
An Ant BuildException has occured: The following error occurred while executing this line:
/home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:59: Unable to create Root Module: config {config/checkstyle_non_main_files_checks.xml}, classpath {/home/rivanov/java/github/checkstyle/checkstyle/target/classes:/home/rivanov/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar:/home/rivanov/.m2/repository/org/antlr/antlr4-runtime/4.7.1/antlr4-runtime-4.7.1.jar:/home/rivanov/.m2/repository/commons-beanutils/commons-beanutils/1.9.3/commons-beanutils-1.9.3.jar:/home/rivanov/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/home/rivanov/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:/home/rivanov/.m2/repository/commons-cli/commons-cli/1.4/commons-cli-1.4.jar:/home/rivanov/.m2/repository/com/google/guava/guava/25.1-jre/guava-25.1-jre.jar:/home/rivanov/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/rivanov/.m2/repository/org/checkerframework/checker-qual/2.0.0/checker-qual-2.0.0.jar:/home/rivanov/.m2/repository/com/google/errorprone/error_prone_annotations/2.1.3/error_prone_annotations-2.1.3.jar:/home/rivanov/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/home/rivanov/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar:/home/rivanov/.m2/repository/net/sf/saxon/Saxon-HE/9.8.0-12/Saxon-HE-9.8.0-12.jar}.
around Ant part ...<ant antfile="config/ant-phase-verify.xml"/>... @ 8:47 in /home/rivanov/java/github/checkstyle/checkstyle/target/antrun/build-main.xml (org.apache.maven.plugins:maven-antrun-plugin:1.8:run:ant-phase-verify:verify)

org.apache.maven.plugin.MojoExecutionException: An Ant BuildException has occured: The following error occurred while executing this line:
/home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:59: Unable to create Root Module: config {config/checkstyle_non_main_files_checks.xml}, classpath {/home/rivanov/java/github/checkstyle/checkstyle/target/classes:/home/rivanov/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar:/home/rivanov/.m2/repository/org/antlr/antlr4-runtime/4.7.1/antlr4-runtime-4.7.1.jar:/home/rivanov/.m2/repository/commons-beanutils/commons-beanutils/1.9.3/commons-beanutils-1.9.3.jar:/home/rivanov/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/home/rivanov/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:/home/rivanov/.m2/repository/commons-cli/commons-cli/1.4/commons-cli-1.4.jar:/home/rivanov/.m2/repository/com/google/guava/guava/25.1-jre/guava-25.1-jre.jar:/home/rivanov/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/rivanov/.m2/repository/org/checkerframework/checker-qual/2.0.0/checker-qual-2.0.0.jar:/home/rivanov/.m2/repository/com/google/errorprone/error_prone_annotations/2.1.3/error_prone_annotations-2.1.3.jar:/home/rivanov/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/home/rivanov/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar:/home/rivanov/.m2/repository/net/sf/saxon/Saxon-HE/9.8.0-12/Saxon-HE-9.8.0-12.jar}.
around Ant part ...<ant antfile="config/ant-phase-verify.xml"/>... @ 8:47 in /home/rivanov/java/github/checkstyle/checkstyle/target/antrun/build-main.xml
	at org.apache.maven.plugin.antrun.AntRunMojo.execute(AntRunMojo.java:342)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:331)
	at org.eclipse.m2e.core.internal.embedder.MavenImpl$11.call(MavenImpl.java:1362)
	at org.eclipse.m2e.core.internal.embedder.MavenImpl$11.call(MavenImpl.java:1)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:112)
	at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:1360)
	at org.eclipse.m2e.core.project.configurator.MojoExecutionBuildParticipant.build(MojoExecutionBuildParticipant.java:52)
	at org.eclipse.m2e.core.internal.builder.MavenBuilderImpl.build(MavenBuilderImpl.java:137)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$1.method(MavenBuilder.java:172)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$1.method(MavenBuilder.java:1)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod$1$1.call(MavenBuilder.java:115)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:112)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod$1.call(MavenBuilder.java:105)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
	at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:99)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod.execute(MavenBuilder.java:86)
	at org.eclipse.m2e.core.internal.builder.MavenBuilder.build(MavenBuilder.java:200)
	at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
	at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
	at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
	at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
	at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:142)
	at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:232)
	at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: /home/rivanov/java/github/checkstyle/checkstyle/target/antrun/build-main.xml:8: The following error occurred while executing this line:
/home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:59: Unable to create Root Module: config {config/checkstyle_non_main_files_checks.xml}, classpath {/home/rivanov/java/github/checkstyle/checkstyle/target/classes:/home/rivanov/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar:/home/rivanov/.m2/repository/org/antlr/antlr4-runtime/4.7.1/antlr4-runtime-4.7.1.jar:/home/rivanov/.m2/repository/commons-beanutils/commons-beanutils/1.9.3/commons-beanutils-1.9.3.jar:/home/rivanov/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/home/rivanov/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:/home/rivanov/.m2/repository/commons-cli/commons-cli/1.4/commons-cli-1.4.jar:/home/rivanov/.m2/repository/com/google/guava/guava/25.1-jre/guava-25.1-jre.jar:/home/rivanov/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/rivanov/.m2/repository/org/checkerframework/checker-qual/2.0.0/checker-qual-2.0.0.jar:/home/rivanov/.m2/repository/com/google/errorprone/error_prone_annotations/2.1.3/error_prone_annotations-2.1.3.jar:/home/rivanov/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/home/rivanov/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar:/home/rivanov/.m2/repository/net/sf/saxon/Saxon-HE/9.8.0-12/Saxon-HE-9.8.0-12.jar}.
	at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:568)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:443)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
	at org.apache.maven.plugin.antrun.AntRunMojo.execute(AntRunMojo.java:313)
	... 32 more
Caused by: /home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:59: Unable to create Root Module: config {config/checkstyle_non_main_files_checks.xml}, classpath {/home/rivanov/java/github/checkstyle/checkstyle/target/classes:/home/rivanov/.m2/repository/antlr/antlr/2.7.7/antlr-2.7.7.jar:/home/rivanov/.m2/repository/org/antlr/antlr4-runtime/4.7.1/antlr4-runtime-4.7.1.jar:/home/rivanov/.m2/repository/commons-beanutils/commons-beanutils/1.9.3/commons-beanutils-1.9.3.jar:/home/rivanov/.m2/repository/commons-logging/commons-logging/1.2/commons-logging-1.2.jar:/home/rivanov/.m2/repository/commons-collections/commons-collections/3.2.2/commons-collections-3.2.2.jar:/home/rivanov/.m2/repository/commons-cli/commons-cli/1.4/commons-cli-1.4.jar:/home/rivanov/.m2/repository/com/google/guava/guava/25.1-jre/guava-25.1-jre.jar:/home/rivanov/.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar:/home/rivanov/.m2/repository/org/checkerframework/checker-qual/2.0.0/checker-qual-2.0.0.jar:/home/rivanov/.m2/repository/com/google/errorprone/error_prone_annotations/2.1.3/error_prone_annotations-2.1.3.jar:/home/rivanov/.m2/repository/com/google/j2objc/j2objc-annotations/1.1/j2objc-annotations-1.1.jar:/home/rivanov/.m2/repository/org/codehaus/mojo/animal-sniffer-annotations/1.14/animal-sniffer-annotations-1.14.jar:/home/rivanov/.m2/repository/net/sf/saxon/Saxon-HE/9.8.0-12/Saxon-HE-9.8.0-12.jar}.
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.createRootModule(CheckstyleAntTask.java:419)
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.realExecute(CheckstyleAntTask.java:311)
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.execute(CheckstyleAntTask.java:294)
	at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
	at sun.reflect.GeneratedMethodAccessor73.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
	at org.apache.tools.ant.Task.perform(Task.java:348)
	at org.apache.tools.ant.Target.execute(Target.java:435)
	at org.apache.tools.ant.Target.performTasks(Target.java:456)
	at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
	at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
	at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
	at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:441)
	... 44 more
Caused by: com.puppycrawl.tools.checkstyle.api.CheckstyleException: Unable to find: config/checkstyle_non_main_files_checks.xml
	at com.puppycrawl.tools.checkstyle.utils.CommonUtil.getUriByFilename(CommonUtil.java:510)
	at com.puppycrawl.tools.checkstyle.ConfigurationLoader.loadConfiguration(ConfigurationLoader.java:390)
	at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.createRootModule(CheckstyleAntTask.java:397)
	... 58 more
	pom.xml	/checkstyle	line 1201	Maven Build Problem

```

**Comment by romani** [comment]
one more weird problem:
```
Description Resource    Path    Location    Type
An Ant BuildException has occured: The following error occurred while executing this line:
/home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:29: java.lang.Error:
Error was thrown while processing /home/rivanov/java/github/checkstyle/checkstyle/src/it/java/
com/google/checkstyle/test/chapter3filestructure/rule333orderingandspacing/CustomImportOrderTest.java
around Ant part ...<ant antfile="config/ant-phase-verify.xml"/>... @ 8:47 in 
/home/rivanov/java/github/checkstyle/checkstyle/target/antrun/build-main.xml (org.apache.maven.plugins:maven-antrun-plugin:1.8:run:ant-phase-verify:verify)

org.apache.maven.plugin.MojoExecutionException: An Ant BuildException has occured: 
The following error occurred while executing this line:
/home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:29: java.lang.Error: 
Error was thrown while processing /home/rivanov/java/github/checkstyle/checkstyle/src/it/
java/com/google/checkstyle/test/chapter3filestructure/rule333orderingandspacing/CustomImportOrderTest.java
around Ant part ...<ant antfile="config/ant-phase-verify.xml"/>... @ 8:47 in 
/home/rivanov/java/github/checkstyle/checkstyle/target/antrun/build-main.xml
at org.apache.maven.plugin.antrun.AntRunMojo.execute(AntRunMojo.java:342)
at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:331)
at org.eclipse.m2e.core.internal.embedder.MavenImpl$11.call(MavenImpl.java:1362)
at org.eclipse.m2e.core.internal.embedder.MavenImpl$11.call(MavenImpl.java:1)
at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:112)
at org.eclipse.m2e.core.internal.embedder.MavenImpl.execute(MavenImpl.java:1360)
at org.eclipse.m2e.core.project.configurator.MojoExecutionBuildParticipant.build(MojoExecutionBuildParticipant.java:52)
at org.eclipse.m2e.core.internal.builder.MavenBuilderImpl.build(MavenBuilderImpl.java:137)
at org.eclipse.m2e.core.internal.builder.MavenBuilder$1.method(MavenBuilder.java:172)
at org.eclipse.m2e.core.internal.builder.MavenBuilder$1.method(MavenBuilder.java:1)
at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod$1$1.call(MavenBuilder.java:115)
at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:112)
at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod$1.call(MavenBuilder.java:105)
at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.executeBare(MavenExecutionContext.java:177)
at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:151)
at org.eclipse.m2e.core.internal.embedder.MavenExecutionContext.execute(MavenExecutionContext.java:99)
at org.eclipse.m2e.core.internal.builder.MavenBuilder$BuildMethod.execute(MavenBuilder.java:86)
at org.eclipse.m2e.core.internal.builder.MavenBuilder.build(MavenBuilder.java:200)
at org.eclipse.core.internal.events.BuildManager$2.run(BuildManager.java:735)
at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:206)
at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:246)
at org.eclipse.core.internal.events.BuildManager$1.run(BuildManager.java:301)
at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
at org.eclipse.core.internal.events.BuildManager.basicBuild(BuildManager.java:304)
at org.eclipse.core.internal.events.BuildManager.basicBuildLoop(BuildManager.java:360)
at org.eclipse.core.internal.events.BuildManager.build(BuildManager.java:383)
at org.eclipse.core.internal.events.AutoBuildJob.doBuild(AutoBuildJob.java:142)
at org.eclipse.core.internal.events.AutoBuildJob.run(AutoBuildJob.java:232)
at org.eclipse.core.internal.jobs.Worker.run(Worker.java:56)
Caused by: /home/rivanov/java/github/checkstyle/checkstyle/target/antrun/build-main.xml:8: 
The following error occurred while executing this line:
/home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:29: java.lang.Error: 
Error was thrown while processing /home/rivanov/java/github/checkstyle/checkstyle/src/it/
java/com/google/checkstyle/test/chapter3filestructure/rule333orderingandspacing/CustomImportOrderTest.java
at org.apache.tools.ant.ProjectHelper.addLocationToBuildException(ProjectHelper.java:568)
at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:443)
at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
at org.apache.tools.ant.Task.perform(Task.java:348)
at org.apache.tools.ant.Target.execute(Target.java:435)
at org.apache.tools.ant.Target.performTasks(Target.java:456)
at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
at org.apache.tools.ant.Project.executeTarget(Project.java:1364)
at org.apache.maven.plugin.antrun.AntRunMojo.execute(AntRunMojo.java:313)
... 32 more
Caused by: /home/rivanov/java/github/checkstyle/checkstyle/config/ant-phase-verify.xml:29: 
java.lang.Error: Error was thrown while processing /home/rivanov/java/github/checkstyle/
checkstyle/src/it/java/com/google/checkstyle/test/chapter3filestructure/rule333orderingandspacing/CustomImportOrderTest.java
at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:116)
at org.apache.tools.ant.Task.perform(Task.java:348)
at org.apache.tools.ant.Target.execute(Target.java:435)
at org.apache.tools.ant.Target.performTasks(Target.java:456)
at org.apache.tools.ant.Project.executeSortedTargets(Project.java:1393)
at org.apache.tools.ant.helper.SingleCheckExecutor.executeTargets(SingleCheckExecutor.java:38)
at org.apache.tools.ant.Project.executeTargets(Project.java:1248)
at org.apache.tools.ant.taskdefs.Ant.execute(Ant.java:441)
... 44 more
Caused by: java.lang.Error: Error was thrown while processing /home/rivanov/java/github/
checkstyle/checkstyle/src/it/java/com/google/checkstyle/test/chapter3filestructure/
rule333orderingandspacing/CustomImportOrderTest.java
at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:299)
at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:213)
at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.processFiles(CheckstyleAntTask.java:353)
at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.realExecute(CheckstyleAntTask.java:322)
at com.puppycrawl.tools.checkstyle.ant.CheckstyleAntTask.execute(CheckstyleAntTask.java:294)
at org.apache.tools.ant.UnknownElement.execute(UnknownElement.java:292)
at sun.reflect.GeneratedMethodAccessor97.invoke(Unknown Source)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at org.apache.tools.ant.dispatch.DispatchUtils.execute(DispatchUtils.java:106)
... 51 more
Caused by: java.lang.NoSuchMethodError: com.google.common.base.CharMatcher.whitespace()
Lcom/google/common/base/CharMatcher;
at com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck.containsForbiddenFragment(SummaryJavadocCheck.java:260)
at com.puppycrawl.tools.checkstyle.checks.javadoc.SummaryJavadocCheck.visitJavadocToken(SummaryJavadocCheck.java:147)
at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck.walk(AbstractJavadocCheck.java:351)
at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck.processTree(AbstractJavadocCheck.java:336)
at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck.visitToken(AbstractJavadocCheck.java:303)
at com.puppycrawl.tools.checkstyle.TreeWalker.notifyVisit(TreeWalker.java:389)
at com.puppycrawl.tools.checkstyle.TreeWalker.processIter(TreeWalker.java:500)
at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:329)
at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:186)
at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:81)
at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:316)
at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:286)
... 60 more
pom.xml /checkstyle line 1201   Maven Build Problem

```

**Comment by rnveach** [comment]
> An Ant BuildException has occured: 

These have also been around since I have been working on checkstyle.

**Comment by romani** [comment]
> These have also been around since I have been working on checkstyle.

It is better to keep track of problems.
One day it will be resolved or workaround provided. At least users can know that it is exists , known problem and it non-critical, could be ignored.
ERRORs are never should be ignored.

**Comment by romani** [comment]
eclipse compiler from command line have no problems with different package and folder BUT Eclipse UI mark it as ERROR, .... I do not know how to reproduce this in command line, so enforcing of PackageDeclarationCheck is required for now.
 
All options -  https://help.eclipse.org/neon/index.jsp?topic=%2Forg.eclipse.jdt.doc.user%2Ftasks%2Ftask-using_batch_compiler.htm

**Comment by romani** [comment]
Original issue with problems to find m2 connectors is not reproducible any more (even when I tried on newly unpacked eclipse).  Looks like pressing "Resolve All Later" worked and it was temporal problem with connection to Eclipse web site to get connectors.

Steps:
1) Import checkstyle project to Eclipse (there will be some errors and warnings)
2) Close Eclipse
3) `cp config/org.eclipse.jdt.core.prefs .settings/`
4) open Eclipse "Build Path > Configure Build Path..."
update as show at http://checkstyle.sourceforge.net/images/gui_screenshot_eclipse_ignore_warnings.png
5) There will stay only warnings, if you do not like them, you can suppress them by https://github.com/checkstyle/checkstyle/issues/5916#issuecomment-397141522
